src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/check_health.py

```diff
@@ -0,0 +1,428 @@
+        helm_command = get_helm_command()
+    except CLIError:
+        accumulated_errors |= _error_handler(Errors.HELM_COMMAND_ERROR,
+                                             "Obtain helm command",
+                                             ignore_errors,
+                                             "HELM OPERATIONS FAILED")
+        return accumulated_errors
+
+    # Helm check
+    output, _ = _subprocess_communicate([helm_command, "version"])
+
+    lines = output.splitlines()
+    regex = re.compile(r'SemVer:"([0-9.vb]+)"')
+    client_version = ""
+    server_version = ""
+
+    for line in lines:
+        if line.startswith('Client'):
+            try:
+                client_version = regex.search(line).group(1)
+            except AttributeError:
+                pass
+        elif line.startswith('Server'):
+            try:
+                server_version = regex.search(line).group(1)
+            except AttributeError:
+                pass
+
+    if client_version in ("", None):
+        accumulated_errors |= _error_handler(Errors.HELM_CLIENT_VERSION_ERROR, "Helm client version", ignore_errors)
+    else:
+        print('Helm client version: {}'.format(client_version))
+
+    if server_version in ("", None):
+        accumulated_errors |= _error_handler(Errors.HELM_SERVER_VERSION_ERROR, "Helm server version", ignore_errors)
+    else:
+        print('Helm server version: {}'.format(server_version))
+
+    return accumulated_errors
+
+
+def _check_health_environment(ignore_errors):
+    accumulated_errors = _get_docker_status_and_version(ignore_errors)
+    accumulated_errors |= _get_cli_version()
+    accumulated_errors |= _get_helm_version(ignore_errors)
+
+    return accumulated_errors
+
+
+## Checks for the connectivity ##
+
+
+def _get_registry_status(login_server, ignore_errors):
+    accumulated_errors = 0
+
+    # Check DNS lookup
+    import socket
+
+    registry_ip = ""
+
+    try:
+        registry_ip = socket.gethostbyname(login_server)
+    except socket.gaierror:
+        pass
+
+    if registry_ip == "":
+        accumulated_errors |= _error_handler(Errors.CONNECTIVITY_REGISTRY_STATUS_ERROR,
+                                             "DNS lookup to {}".format(login_server),
+                                             ignore_errors,
+                                             "CONNECTIVITY OPERATIONS FAILED")
+    else:
+        print_pass("DNS lookup to {} at IP {}".format(login_server, registry_ip))
+
+    return accumulated_errors
+
+
+def _get_endpoint_and_token_status(cmd, login_server, ignore_errors):
+    accumulated_errors = 0
+
+    import requests
+    try:
+        from urllib.parse import urlencode, urlparse, urlunparse
+    except ImportError:
+        from urllib import urlencode
+        from urlparse import urlparse, urlunparse
+
+    # Check access to login endpoint
+    url = 'https://' + login_server + '/v2/'
+    challenge = requests.get(url)
+
+    if challenge.status_code == 403:
```

Comment:
The following code has a lot of duplication with https://github.com/Azure/azure-cli/blob/f8cae0185ac413c5351804409b829e4ec1ae560f/src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_docker_utils.py#L44

Could you try to see if you can reuse the existing code? Your logic can be added to existing code too to provide additional error messages.